internal/dwarf/frame/table.go

```diff
@@ -531,135 +537,150 @@ func restoreextended(frameContext *FrameContext) {
 }
 
-func offsetextendedsf(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func offsetextendedsf(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _    = util.DecodeULEB128(frameContext.buf)
-		offset, _ = util.DecodeSLEB128(frameContext.buf)
+		reg, _    = util.DecodeULEB128(ctx.buf)
+		offset, _ = util.DecodeSLEB128(ctx.buf)
 	)
 
 	// fmt.Println("register", reg, "set to dwrule")
 	frame.Regs[reg] = DWRule{Offset: offset * frame.dataAlignment, Rule: RuleOffset}
 }
 
-func valoffset(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func valoffset(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _    = util.DecodeULEB128(frameContext.buf)
-		offset, _ = util.DecodeULEB128(frameContext.buf)
+		reg, _    = util.DecodeULEB128(ctx.buf)
+		offset, _ = util.DecodeULEB128(ctx.buf)
 	)
 
 	// fmt.Println("register", reg, "set to dwrule")
 	frame.Regs[reg] = DWRule{Offset: int64(offset), Rule: RuleValOffset}
 }
 
-func valoffsetsf(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func valoffsetsf(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _    = util.DecodeULEB128(frameContext.buf)
-		offset, _ = util.DecodeSLEB128(frameContext.buf)
+		reg, _    = util.DecodeULEB128(ctx.buf)
+		offset, _ = util.DecodeSLEB128(ctx.buf)
 	)
 
 	// fmt.Println("register", reg, "set to offset")
 	frame.Regs[reg] = DWRule{Offset: offset * frame.dataAlignment, Rule: RuleValOffset}
 }
 
-func valexpression(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func valexpression(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _ = util.DecodeULEB128(frameContext.buf)
-		l, _   = util.DecodeULEB128(frameContext.buf)
-		expr   = frameContext.buf.Next(int(l))
+		reg, _ = util.DecodeULEB128(ctx.buf)
+		l, _   = util.DecodeULEB128(ctx.buf)
+		expr   = ctx.buf.Next(int(l))
 	)
 
 	frame.Regs[reg] = DWRule{Rule: RuleValExpression, Expression: expr}
 }
 
-func louser(frameContext *FrameContext) {
-	frameContext.buf.Next(1)
+func louser(ctx *Context) {
+	ctx.buf.Next(1)
 }
 
-func hiuser(frameContext *FrameContext) {
-	frameContext.buf.Next(1)
+func hiuser(ctx *Context) {
+	ctx.buf.Next(1)
 }
 
-func gnuargsize(frameContext *FrameContext) {
+// SPARC-specific.
+func gnuwindowsave(ctx *Context) {
+	// Skip, do nothing. Architecture is not supported.
+	_, _ = util.DecodeSLEB128(ctx.buf)
+}
 
+func gnuargsize(ctx *Context) {
 	// The DW_CFA_GNU_args_size instruction takes an unsigned LEB128 operand representing an argument size.
-	// Just read and do nothing.
-	// TODO(kakkoyun): Implement this.
-	_, _ = util.DecodeSLEB128(frameContext.buf)
+	// This instruction specifies the total of the size of the arguments which have been pushed onto the stack.
+	frame := ctx.getCurrentInstruction()
+	size, _ := util.DecodeULEB128(ctx.buf)
+	frame.ArgsSize = size
```

Comment:
This doesn't change anything for stack unwinding. I decided to store it just in case if we want to push changes to the upstream.